util/progress/reader.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
```

Comment:
Rather than rolling our own progress bar, can we use an existing, well-tested one? https://github.com/schollz/progressbar looks popular.

For code size purposes, we will probably also want to build tag any progress-bar-related stuff, so that it doesn't get compiled in on mobile, where it will never be used.

Reply:
Yep, I'm happy to switch to a third-party dependency if you'd prefer that.  Do you want me to push a commit using that library?